### ISSUE TITLE
Clink v1.0.0.a1 - `-s|--scripts [path]` command line arg removed?

### ISSUE THREAD

**Issue by daxgames** [mridgers/clink#519]
[Cmder](https://github.com/cmderdev/cmder) team is attempting to integrate Clink 1.0.0 Alphas and getting this error upon starting Cmder:

```
inject: invalid option -- `-s'
Running Git for Windows one time Post Install....
"running post-install"
mkdir: cannot change permissions of ‘/dev/shm’: Permission denied
mkdir: cannot change permissions of ‘/dev/mqueue’: Permission denied
'C:\WINDOWS\system32\drivers\etc\hosts' -> '/etc/hosts'
'C:\WINDOWS\system32\drivers\etc\protocol' -> '/etc/protocols'
'C:\WINDOWS\system32\drivers\etc\services' -> '/etc/services'
'C:\WINDOWS\system32\drivers\etc\networks' -> '/etc/networks'
The batch file cannot be found.
Elapsed Time: 0:0:17.04 (17.04s total)

C:\Users\dtgam\cmder dev>Injecting Clink! inject: invalid option -- `-s'
```

Cmder relies on this argument to load extra lua from `%cmder_root%\vendor` so we can have a package default lua config while setting the `--profile` to `%cmder_root%\config`  for user unique lua config using the following startup code:

```
:: Pick right version of clink
if "%PROCESSOR_ARCHITECTURE%"=="x86" (
    set architecture=86
    set architecture_bits=32
) else (
    set architecture=64
    set architecture_bits=64
)

if "%CMDER_CLINK%" == "1" (
  %lib_console% verbose_output "Injecting Clink!"

  :: Run clink
  if defined CMDER_USER_CONFIG (
    if not exist "%CMDER_USER_CONFIG%\settings" (
      echo Generating clink initial settings in "%CMDER_USER_CONFIG%\settings"
      copy "%CMDER_ROOT%\vendor\clink_settings.default" "%CMDER_USER_CONFIG%\settings"
      echo Additional *.lua files in "%CMDER_USER_CONFIG%" are loaded on startup.\
    )
    "%CMDER_ROOT%\vendor\clink\clink_x%architecture%.exe" inject --quiet --profile "%CMDER_USER_CONFIG%" --scripts "%CMDER_ROOT%\vendor"
  ) else (
    if not exist "%CMDER_ROOT%\config\settings" (
      echo Generating clink initial settings in "%CMDER_ROOT%\config\settings"
      copy "%CMDER_ROOT%\vendor\clink_settings.default" "%CMDER_ROOT%\config\settings"
      echo Additional *.lua files in "%CMDER_ROOT%\config" are loaded on startup.
    )
    "%CMDER_ROOT%\vendor\clink\clink_x%architecture%.exe" inject --quiet --profile "%CMDER_ROOT%\config" --scripts "%CMDER_ROOT%\vendor"
  )
) else (
  %lib_console% verbose_output "WARNING: Incompatible 'ComSpec/Shell' Detetected Skipping Clink Injection!"
)
```
As you can see we use the `--scripts [path]` argument to point at an additional config dir for including additional lua code.  Startup with Clink 0.4.9 results in:

```
Running Git for Windows one time Post Install....
"running post-install"
mkdir: cannot change permissions of ‘/dev/shm’: Permission denied
mkdir: cannot change permissions of ‘/dev/mqueue’: Permission denied
'C:\WINDOWS\system32\drivers\etc\hosts' -> '/etc/hosts'
'C:\WINDOWS\system32\drivers\etc\protocol' -> '/etc/protocols'
'C:\WINDOWS\system32\drivers\etc\services' -> '/etc/services'
'C:\WINDOWS\system32\drivers\etc\networks' -> '/etc/networks'
Elapsed Time: 0:0:17.04 (17.04s total)

C:\Users\dtgam\cmder dev(conemu_args)λ
```

1.  Why was this removed?

2. Can it be added back?

**Comment by chrisant996** [comment]
Makes sense.  I'll add that back in the [chrisant996/clink](https://github.com/chrisant996/clink) fork.

**Comment by daxgames** [comment]
@chrisant996 thanks.

**Comment by chrisant996** [comment]
Existing behavior:
- **v0.4.9**: uses either the `inject --scripts` flag **or** the clink dll's directory to load scripts.
- **v1.0.0a1**:  uses the `clink.path` setting **and** the `clink_path` environment variable to load scripts.  And if both are empty (which is the default), then it doesn't load any users scripts (just the embedded scripts).

The `--scripts` flag was meant for internal development purposes, as a debugging convenience.  Relying on that flag means the only way to get the desired behavior is by launching `clink.bat inject --scripts {path}`, so it can interfere with launching through any of the other mechanisms.

**Can cmder use the `clink_path` environment variable instead?**  That would accomplish the same thing as `--scripts` in the newer clink versions, but would be more robust and would also affect `clink set`.  (The flag doesn't affect `clink set`, so v0.4.9 can report settings inaccurately, since a different set of lua scripts may run.)

<br>
<br>

Or if not, then here's an alternative strategy I could implement in the [chrisant996/clink](https://github.com/chrisant996/clink) fork:
1. If `inject --scripts` is used then load only those scripts, and ignore the `clink.path` setting and `clink_path` environment variables (otherwise `--scripts` could easily run the same lua scripts multiple times).
2. If neither the `clink.path` setting _nor_ the `clink_path` environment variable are set then load scripts from the clink dll's directory (default behavior would be the same as 0.4.9).
3. If `clink.path` and/or `clink_path` are set then load scripts from those directories.  Each is a semicolon-delimited list of paths.

**Comment by daxgames** [comment]
I don't see why we couldn't use clink.path setting  or clink_path env var.

**Comment by chrisant996** [comment]
> I don't see why we couldn't use clink.path setting or clink_path env var.

Between the two, I would lean towards `clink_path` env var, because it's standalone -- i.e. the `clink_path` setting depends on the profile directory.

But the main reason I can see for using the `--scripts` flag is that once it's been injected, there's no way for subsequent config changes to interfere with the cmder scripts while the cmd/clink process is running.  Using the setting or env var could both be overwritten by the user and break the cmder/clink experience, with no obvious way for the user to fix it.

Because of that, I went ahead and implemented the entire alternative strategy I outlined above.
